### DISCUSSION TITLE
[Enhancement]: Entering orders manually should allow automatic shipping and tax calculations.

### DISCUSSION THREAD

**Discussion by mark-orion** [woocommerce/woocommerce#47287] [General]
### Describe the solution you'd like

When I enter an order manually, then it should be possible to have the tax and shipping calculated automatically in the same manner as it happens at checkout. Manual ordering needs optimising for a quick workflow.
Even the "Add Items" button should be replaced with separate buttons for adding products, shipping, tax and fees.
The current implementation is too limiting and it needs to many clicks to enter an order.

**Comment by gmo81** [comment]
I agree. This has been a long term issue in Woocommerce when creating orders manually, and one that has bitten us countless times. Several other people too.

The typical explanation for why this must be manual doesn’t make sense to me: ‘The country could be changed after the order is entered’.

It is true, it could be - especially if you don’t put the end customer in first. BUT - this is not the normal case. I believe things should be optimised for the expected/normal case, not the corner case. 

The ‘recalculate’ function could be used in this non-standard case (if the base location, and therefore, potentially tax implication for the order has changed), but I have another suggestion (or several really) that I think solve this problem either way:

1. Although I haven’t dived into the code, it seems observably that the current recalculate functionality is an AJAX call - the page itself is not reloaded. 
Potential solution with this in mind: Following a product addition, quantity or price change, addition of fee, or addition of shipping - call the recalculate function automatically. This could also fire (if the order is not $0), if the customer or customer location is changed. The function obvioulsy already exists, so simply add the calling of it, to the use cases where something changes.

2. As recalculation of taxes is based on the customer location, if not automatically done with an AJAX call (as ‘recalculate’ works today), at minimum, do the recalculation on ‘save’ of the manual order.

Both solutions could apply the same logic to adding a shipping fee.

My suggestion is to implement both. Live recalculation of taxes (and potentially shipping) using the existing function that the recalculate button fires - just fire it automatically on change.

And for good measure, run the recalculation function on save of a manual order (just in case, on the specific user’s browser, the Javascript does not or cannot fire due to say some sort of browser restrictions stopping AJAX from operating).

This way, you remove the unnecessary, and often missed step of hitting ‘recalculate’ on a manual order, and don’t end up sending tax invoices to customers that are missing tax. Sometimes they aren’t so keen to pay for a reissued invoice with tax following manual ‘human error’, but this task of calculating tax should (at least optionally) not be left to humans at all in my opinion.

If you are concerned about the behaviour clobbering someone’s workflow, I would suggest considering:
a. Make this an option Woocommerce users can turn on (and leave it off for existing stores upgrading)
b. If someone hits recalculate already, and there is no tax, the recalculate adds no tax - so functionally, in theory, it shouldn’t really break someone’s workflow even if not optional - but for safety, as this changes a behaviour, I would suggest having a toggle switch in the woocommerce settings per (A) anyway.

Happy to provide further information, or even assistance, code review, whatever for this change - it would really solve probably the most painful thing we see today using Woocommerce and dealing with many orders we have to put in manually. Would really appreciate the help on this as it has plagued us for over seven years.

**Comment by coreymckrill** [comment]
This is definitely a topic worthy of discussion. I'm going to officially convert this issue to a "discussion" so that it's more discoverable.
